src/import-data/ImportData.ts

```diff
@@ -1,72 +1,88 @@
-// * Reading from the URL
-// * Loading initial Data
-// * Managing the import
-export const ImportData = () => {
+const getTitleForImportRequest = (importRequest: ImportRequest): string => {
+  if (importRequest.type === 'tdr-snapshot-export') {
+    return `Importing snapshot ${importRequest.snapshotName}`;
+  }
+
+  if (
+    importRequest.type === 'tdr-snapshot-reference' ||
+    importRequest.type === 'catalog-dataset' ||
+    importRequest.type === 'catalog-snapshots'
+  ) {
+    return 'Linking data to a workspace';
+  }
+
+  return 'Importing data to a workspace';
+};
+
+export interface ImportDataProps {
+  importRequest: ImportRequest;
+}
+
+export const ImportData = (props: ImportDataProps): ReactNode => {
+  const { importRequest } = props;
   const {
-    query: {
-      url,
-      format,
-      ad,
-      wid,
-      template,
-      snapshotId,
-      snapshotName,
-      snapshotIds,
-      referrer,
-      tdrmanifest,
-      catalogDatasetId,
-      tdrSyncPermissions,
-    },
+    query: { ad, format, wid, template },
   } = Nav.useRoute();
   const [templateWorkspaces, setTemplateWorkspaces] = useState<{ [key: string]: TemplateWorkspaceInfo[] }>();
   const [userHasBillingProjects, setUserHasBillingProjects] = useState(true);
   const [snapshotResponses, setSnapshotResponses] = useState<{ status: string; message: string | undefined }[]>();
   const [isImporting, setIsImporting] = useState(false);
 
   const { dataCatalog } = useDataCatalog();
-  const snapshots = _.flow(
-    _.filter(
-      (snapshot: Dataset) => !!snapshot['dct:identifier'] && _.includes(snapshot['dct:identifier'], snapshotIds)
-    ),
-    _.map((snapshot) => ({
-      // The previous step filters the list to only datasets with 'dct:identifier' defined
-      id: snapshot['dct:identifier']!,
-      title: snapshot['dct:title'],
-      description: snapshot['dct:description'],
-    }))
-  )(dataCatalog);
+  const snapshots =
+    importRequest.type === 'catalog-snapshots'
+      ? _.flow(
+          _.filter(
+            (snapshot: Dataset) =>
+              !!snapshot['dct:identifier'] && _.includes(snapshot['dct:identifier'], importRequest.snapshotIds)
+          ),
+          _.map((snapshot) => ({
+            // The previous step filters the list to only datasets with 'dct:identifier' defined
+            id: snapshot['dct:identifier']!,
+            title: snapshot['dct:title'],
+            description: snapshot['dct:description'],
+          }))
+        )(dataCatalog)
+      : [];
 
   const isDataset = !_.includes(format, ['snapshot', 'tdrexport']);
-  const header = Utils.cond(
-    [referrer === 'data-catalog', () => 'Linking data to a workspace'],
-    [isDataset, () => `Dataset ${snapshotName}`],
-    [Utils.DEFAULT, () => `Snapshot ${snapshotName}`]
-  );
 
-  const isProtectedData = isProtectedSource(url, format);
+  const isProtectedData = importRequest.type === 'pfb' && isProtectedSource(importRequest.url);
```

Comment:
`isProtectedSource` is currently only valid for PFB imports. I expect to clean this up some more in future PRs that focus on protected data.

Reply:
In the meantime you could change `isProtectedSource()` to accept `PFBImportRequest`
```suggestion
  const isProtectedData = importRequest.type === 'pfb' && isProtectedSource(importRequest);
```

Or make `isProtectedSource` a property / method on `PFBImportRequest`

Reply:
To limit the scope of this PR, I'm going to defer changes to this function to future PRs. Right now, I'm thinking that `isProtectedSource` will end up taking an `ImportRequest` since we'll have to handle protected TDR snapshots.